Flaky Test: TestStreamFailureHeaderPhaseDeny

```
--- FAIL: Test (0.69s)
    --- FAIL: Test/StreamFailureHeaderPhaseDeny (0.03s)
        tlogger.go:133: INFO server.go:734 [core] [Server #468] Server created  (t=+213.125µs)
        tlogger.go:133: INFO server.go:930 [core] [Server #468 ListenSocket #469] ListenSocket created  (t=+325.891µs)
        tlogger.go:133: INFO server.go:734 [core] [Server #470] Server created  (t=+437.736µs)
        tlogger.go:133: INFO server.go:930 [core] [Server #470 ListenSocket #471] ListenSocket created  (t=+550.012µs)
        tlogger.go:133: INFO server.go:734 [core] [Server #472] Server created  (t=+659.683µs)
        tlogger.go:133: INFO server.go:930 [core] [Server #472 ListenSocket #473] ListenSocket created  (t=+798.668µs)
        tlogger.go:133: INFO clientconn.go:1823 [core] original dial target is: "xds:///test-service"  (t=+1.42319ms)
        tlogger.go:133: INFO clientconn.go:514 [core] [Channel #474] Channel created for target "xds:///test-service"  (t=+1.440896ms)
        tlogger.go:133: INFO clientconn.go:245 [core] [Channel #474] parsed dial target is: resolver.Target{URL:url.URL{Scheme:"xds", Opaque:"", User:(*url.Userinfo)(nil), Host:"", Path:"/test-service", Fragment:"", RawQuery:"", RawPath:"", RawFragment:"", ForceQuery:false, OmitHost:false}}  (t=+1.459734ms)
        tlogger.go:133: INFO clientconn.go:246 [core] [Channel #474] Channel authority set to "test-service"  (t=+1.473314ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #474] Channel Connectivity change to CONNECTING  (t=+1.516137ms)
        tlogger.go:133: INFO pool.go:309 [xds] xDS node ID: aa24c8dc-da7e-4686-a1ef-bb1f695c367e  (t=+1.685747ms)
        tlogger.go:133: INFO xds_resolver.go:158 [xds] [xds-resolver 0x2480d2bc6fc0] Creating resolver for target: xds:///test-service  (t=+1.706178ms)
        tlogger.go:133: INFO clientconn.go:1823 [core] original dial target is: "passthrough:///127.0.0.1:45847"  (t=+1.812645ms)
        tlogger.go:133: INFO clientconn.go:514 [core] [Channel #475] Channel created for target "passthrough:///127.0.0.1:45847"  (t=+1.835248ms)
        tlogger.go:133: INFO clientconn.go:245 [core] [Channel #475] parsed dial target is: resolver.Target{URL:url.URL{Scheme:"passthrough", Opaque:"", User:(*url.Userinfo)(nil), Host:"", Path:"/127.0.0.1:45847", Fragment:"", RawQuery:"", RawPath:"", RawFragment:"", ForceQuery:false, OmitHost:false}}  (t=+1.852053ms)
        tlogger.go:133: INFO clientconn.go:246 [core] [Channel #475] Channel authority set to "127.0.0.1:45847"  (t=+1.864171ms)
        tlogger.go:133: INFO clientconn.go:419 [core] [Channel #474] Channel exiting idle mode  (t=+1.911571ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #475] Channel Connectivity change to CONNECTING  (t=+1.987473ms)
        tlogger.go:133: INFO balancer_wrapper.go:121 [core] [Channel #475] Channel switches to new LB policy "pick_first"  (t=+2.032138ms)
        tlogger.go:133: INFO clientconn.go:922 [core] [Channel #475 SubChannel #476] Subchannel created  (t=+2.067531ms)
        tlogger.go:133: INFO clientconn.go:419 [core] [Channel #475] Channel exiting idle mode  (t=+2.098576ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #475 SubChannel #476] Subchannel Connectivity change to CONNECTING  (t=+2.130023ms)
        tlogger.go:133: INFO clientconn.go:1470 [core] [Channel #475 SubChannel #476] Subchannel picks a new address "127.0.0.1:45847" to connect  (t=+2.156222ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #475 SubChannel #476] Subchannel Connectivity change to READY  (t=+2.589ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #475] Channel Connectivity change to READY  (t=+2.635318ms)
        tlogger.go:133: INFO clientconn.go:1823 [core] original dial target is: "127.0.0.1:35831"  (t=+3.579111ms)
        tlogger.go:133: INFO clientconn.go:514 [core] [Channel #479] Channel created for target "127.0.0.1:35831"  (t=+3.609336ms)
        tlogger.go:133: INFO clientconn.go:245 [core] [Channel #479] parsed dial target is: resolver.Target{URL:url.URL{Scheme:"dns", Opaque:"", User:(*url.Userinfo)(nil), Host:"", Path:"/127.0.0.1:35831", Fragment:"", RawQuery:"", RawPath:"", RawFragment:"", ForceQuery:false, OmitHost:false}}  (t=+3.628674ms)
        tlogger.go:133: INFO clientconn.go:246 [core] [Channel #479] Channel authority set to "127.0.0.1:35831"  (t=+3.641763ms)
        tlogger.go:133: INFO balancer_wrapper.go:121 [core] [Channel #474] Channel switches to new LB policy "xds_cluster_manager_experimental"  (t=+3.718627ms)
        tlogger.go:133: INFO clustermanager.go:58 [xds] [xds-cluster-manager-lb 0x2480d28e53e0] Created  (t=+3.74083ms)
        tlogger.go:133: INFO balancergroup.go:274 [xds] [xds-cluster-manager-lb 0x2480d28e53e0] Adding child policy of type "cds_experimental" for child "cluster:cluster-test-service"  (t=+3.798294ms)
        tlogger.go:133: INFO balancergroup.go:102 [xds] [xds-cluster-manager-lb 0x2480d28e53e0] Creating child policy of type "cds_experimental" for child "cluster:cluster-test-service"  (t=+3.811364ms)
        tlogger.go:133: INFO cdsbalancer.go:90 [xds] [cds-lb 0x2480d2413c00] Created  (t=+3.826686ms)
        tlogger.go:133: INFO cdsbalancer.go:167 [xds] [cds-lb 0x2480d2413c00] Received balancer config update: {
              "LoadBalancingConfig": null,
              "cluster": "cluster-test-service",
              "isDynamic": false
            }  (t=+3.846195ms)
        tlogger.go:133: INFO balancer.go:76 [xds] [priority-lb 0x2480d2b0c2a0] Created  (t=+3.900535ms)
        tlogger.go:133: INFO balancergroup.go:528 [xds] [xds-cluster-manager-lb 0x2480d28e53e0] Balancer state update from child cluster:cluster-test-service, new state: {ConnectivityState:CONNECTING Picker:0x2480d2b12980}  (t=+4.03948ms)
        tlogger.go:133: INFO balancerstateaggregator.go:147 [xds] [xds-cluster-manager-lb 0x2480d28e53e0] State update from sub-balancer "cluster:cluster-test-service": {ConnectivityState:CONNECTING Picker:0x2480d2b12980}  (t=+4.07307ms)
        tlogger.go:133: INFO balancergroup.go:274 [xds] [priority-lb 0x2480d2b0c2a0] Adding child policy of type "outlier_detection_experimental" for child "priority-0-0"  (t=+4.084917ms)
        tlogger.go:133: INFO balancergroup.go:102 [xds] [priority-lb 0x2480d2b0c2a0] Creating child policy of type "outlier_detection_experimental" for child "priority-0-0"  (t=+4.096955ms)
        tlogger.go:133: INFO balancer.go:94 [xds] [outlier-detection-lb 0x2480d2b1f1d0] Created  (t=+4.114741ms)
        tlogger.go:133: INFO clusterimpl.go:88 [xds] [xds-cluster-impl-lb 0x2480d27e0780] Created  (t=+4.133719ms)
        tlogger.go:133: INFO clusterimpl.go:100 [xds] [xds-cluster-impl-lb 0x2480d27e0780] xDS credentials in use: false  (t=+4.147359ms)
        tlogger.go:133: INFO weightedtarget.go:65 [xds] [weighted-target-lb 0x2480d28e5aa0] Created  (t=+4.199126ms)
        tlogger.go:133: INFO balancer.go:88 [xds] [wrrlocality-lb 0x2480d274eb10] Created  (t=+4.21567ms)
        tlogger.go:133: INFO balancergroup.go:274 [xds] [weighted-target-lb 0x2480d28e5aa0] Adding child policy of type "round_robin" for child "{region=\"region-1\", zone=\"zone-1\", sub_zone=\"subzone-1\"}"  (t=+5.633552ms)
        tlogger.go:133: INFO balancergroup.go:102 [xds] [weighted-target-lb 0x2480d28e5aa0] Creating child policy of type "round_robin" for child "{region=\"region-1\", zone=\"zone-1\", sub_zone=\"subzone-1\"}"  (t=+5.66578ms)
        tlogger.go:133: INFO roundrobin.go:56 [roundrobin] [0x2480d2ac2db0] Created  (t=+5.691508ms)
        tlogger.go:133: INFO clientconn.go:922 [core] [Channel #474 SubChannel #480] Subchannel created  (t=+5.74747ms)
        tlogger.go:133: INFO balancergroup.go:528 [xds] [weighted-target-lb 0x2480d28e5aa0] Balancer state update from child {region="region-1", zone="zone-1", sub_zone="subzone-1"}, new state: {ConnectivityState:CONNECTING Picker:0x2480d2d330c0}  (t=+5.809552ms)
        tlogger.go:133: INFO aggregator.go:253 [xds] [weighted-target-lb 0x2480d28e5aa0] Child pickers with config: map[{region="region-1", zone="zone-1", sub_zone="subzone-1"}:weight:1,picker:0x2480d2d330c0,state:CONNECTING,stateToAggregate:CONNECTING]  (t=+5.841339ms)
        tlogger.go:133: INFO balancergroup.go:528 [xds] [priority-lb 0x2480d2b0c2a0] Balancer state update from child priority-0-0, new state: {ConnectivityState:CONNECTING Picker:0x2480d23a3d70}  (t=+5.88933ms)
        tlogger.go:133: INFO balancergroup.go:528 [xds] [xds-cluster-manager-lb 0x2480d28e53e0] Balancer state update from child cluster:cluster-test-service, new state: {ConnectivityState:CONNECTING Picker:0x2480d23a3d70}  (t=+5.912093ms)
        tlogger.go:133: INFO balancerstateaggregator.go:147 [xds] [xds-cluster-manager-lb 0x2480d28e53e0] State update from sub-balancer "cluster:cluster-test-service": {ConnectivityState:CONNECTING Picker:0x2480d23a3d70}  (t=+5.929339ms)
        tlogger.go:133: INFO balancerstateaggregator.go:191 [xds] [xds-cluster-manager-lb 0x2480d28e53e0] Child pickers: map[cluster:cluster-test-service:picker:0x2480d23a3d70,state:CONNECTING,stateToAggregate:CONNECTING]  (t=+5.98434ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #479] Channel Connectivity change to CONNECTING  (t=+6.062575ms)
        tlogger.go:133: INFO balancer_wrapper.go:121 [core] [Channel #479] Channel switches to new LB policy "pick_first"  (t=+6.105489ms)
        tlogger.go:133: INFO clientconn.go:922 [core] [Channel #479 SubChannel #481] Subchannel created  (t=+6.143475ms)
        tlogger.go:133: INFO clientconn.go:419 [core] [Channel #479] Channel exiting idle mode  (t=+6.166158ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #474 SubChannel #480] Subchannel Connectivity change to CONNECTING  (t=+6.240277ms)
        tlogger.go:133: INFO clientconn.go:1470 [core] [Channel #474 SubChannel #480] Subchannel picks a new address "localhost:43715" to connect  (t=+6.260397ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #479 SubChannel #481] Subchannel Connectivity change to CONNECTING  (t=+6.796648ms)
        tlogger.go:133: INFO clientconn.go:1470 [core] [Channel #479 SubChannel #481] Subchannel picks a new address "127.0.0.1:35831" to connect  (t=+6.831109ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #479 SubChannel #481] Subchannel Connectivity change to READY  (t=+7.29232ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #479] Channel Connectivity change to READY  (t=+7.372428ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #474] Channel Connectivity change to SHUTDOWN  (t=+7.673732ms)
        tlogger.go:133: INFO resolver_wrapper.go:112 [core] [Channel #474] Closing the name resolver  (t=+7.689376ms)
        tlogger.go:133: INFO balancer_wrapper.go:159 [core] [Channel #474] ccBalancerWrapper: closing  (t=+7.711188ms)
        tlogger.go:133: INFO clusterimpl.go:521 [xds] [xds-cluster-impl-lb 0x2480d27e0780] Shutdown  (t=+7.777496ms)
        tlogger.go:133: INFO balancergroup.go:351 [xds] [priority-lb 0x2480d2b0c2a0] Removing child policy for child "priority-0-0"  (t=+7.803704ms)
        tlogger.go:133: INFO cdsbalancer.go:416 [xds] [cds-lb 0x2480d2413c00] Shutdown  (t=+7.816403ms)
        tlogger.go:133: INFO clustermanager.go:192 [xds] [xds-cluster-manager-lb 0x2480d28e53e0] Shutdown  (t=+7.831555ms)
        tlogger.go:129: WARNING ads_stream.go:410 [xds] [xds-client 0x2480d2929220] [xds-channel 0x2480d2b46be0] [ads-stream 0x2480d29292c0] Sending ADS request for type "type.googleapis.com/envoy.config.listener.v3.Listener", resources: [], version: "1", nonce: "1" failed: EOF  (t=+9.143411ms)
        tlogger.go:129: WARNING ads_stream.go:452 [xds] [xds-client 0x2480d2929220] [xds-channel 0x2480d2b46be0] [ads-stream 0x2480d29292c0] ADS stream closed: rpc error: code = Canceled desc = context canceled  (t=+9.172404ms)
        tlogger.go:133: INFO ads_stream.go:165 [xds] [xds-client 0x2480d2929220] [xds-channel 0x2480d2b46be0] [ads-stream 0x2480d29292c0] Shutdown ADS stream  (t=+9.201256ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #475] Channel Connectivity change to SHUTDOWN  (t=+9.229939ms)
        tlogger.go:133: INFO resolver_wrapper.go:112 [core] [Channel #475] Closing the name resolver  (t=+9.250299ms)
        tlogger.go:133: INFO balancer_wrapper.go:159 [core] [Channel #475] ccBalancerWrapper: closing  (t=+9.265261ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #475 SubChannel #476] Subchannel Connectivity change to SHUTDOWN  (t=+9.31899ms)
        tlogger.go:133: INFO clientconn.go:1696 [core] [Channel #475 SubChannel #476] Subchannel deleted  (t=+9.333992ms)
        tlogger.go:133: INFO clientconn.go:1231 [core] [Channel #475] Channel deleted  (t=+9.456773ms)
        tlogger.go:133: INFO channel.go:143 [xds] [xds-client 0x2480d2929220] [xds-channel 0x2480d2b46be0] Shutdown  (t=+9.486057ms)
        tlogger.go:133: INFO xdsclient.go:210 [xds] [xds-client 0x2480d2929220] Shutdown  (t=+9.515901ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #479] Channel Connectivity change to SHUTDOWN  (t=+9.535229ms)
        tlogger.go:133: INFO resolver_wrapper.go:112 [core] [Channel #479] Closing the name resolver  (t=+9.552595ms)
        tlogger.go:133: INFO balancer_wrapper.go:159 [core] [Channel #479] ccBalancerWrapper: closing  (t=+9.57003ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #479 SubChannel #481] Subchannel Connectivity change to SHUTDOWN  (t=+9.617541ms)
        tlogger.go:133: INFO clientconn.go:1696 [core] [Channel #479 SubChannel #481] Subchannel deleted  (t=+9.635767ms)
        tlogger.go:133: INFO clientconn.go:1231 [core] [Channel #479] Channel deleted  (t=+9.69165ms)
        tlogger.go:133: INFO xds_resolver.go:296 [xds] [xds-resolver 0x2480d2bc6fc0] Shutdown  (t=+9.711089ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #474 SubChannel #480] Subchannel Connectivity change to SHUTDOWN  (t=+9.738529ms)
        tlogger.go:133: INFO clientconn.go:1696 [core] [Channel #474 SubChannel #480] Subchannel deleted  (t=+9.751558ms)
        tlogger.go:133: INFO clientconn.go:1231 [core] [Channel #474] Channel deleted  (t=+9.78642ms)
        tlogger.go:133: INFO server.go:866 [core] [Server #470 ListenSocket #471] ListenSocket deleted  (t=+9.835021ms)
        tlogger.go:133: INFO server.go:866 [core] [Server #472 ListenSocket #473] ListenSocket deleted  (t=+9.873799ms)
        tlogger.go:129: WARNING clientconn.go:1533 [core] [Channel #474 SubChannel #480] grpc: addrConn.createTransport failed to connect to {Addr: "localhost:43715", ServerName: "test-service", Attributes: {"<%!p(xds.hostnameKey={})>": "" , "<%!p(xds.handshakeAttrKey={})>": "<0x2480d27e07f0>" , "<%!p(xds.handshakeClusterNameKey={})>": "cluster-test-service" }, BalancerAttributes: {"<%!p(weight.attributeKey={})>": "<%!p(weight.EndpointInfo={2147483648})>" , "<%!p(wrrlocality.attributeKey={})>": "Locality Weight: 1" , "<%!p(xds.localityKeyType=grpc.xds.internal.address.locality)>": "<%!p(clients.Locality={region-1 zone-1 subzone-1})>" , "<%!p(hierarchy.pathKeyType=grpc.internal.address.hierarchical_path)>": "<0x2480d28e57a0>" }}. Err: connection error: desc = "transport: Error while dialing: dial tcp: lookup localhost: operation was canceled"  (t=+9.99008ms)
        tlogger.go:133: INFO server.go:866 [core] [Server #468 ListenSocket #469] ListenSocket deleted  (t=+10.060154ms)
        --- FAIL: Test/StreamFailureHeaderPhaseDeny/AbruptStreamFailure (0.01s)
            stubserver.go:300: Started test service backend at "127.0.0.1:43715"
            setup.go:45: Created new snapshot cache...
            setup.go:45: Registered Aggregated Discovery Service (ADS)...
            setup.go:45: xDS management server serving at: 127.0.0.1:45847...
            server.go:229: Created new resource snapshot...
            logging.go:30: setting snapshot for node aa24c8dc-da7e-4686-a1ef-bb1f695c367e
            server.go:235: Updated snapshot cache with resource snapshot...
            logging.go:30: respond type.googleapis.com/envoy.config.listener.v3.Listener (requested [test-service]) version "" with version "1" and resources [test-service]
            logging.go:30: open watch 1 for type.googleapis.com/envoy.config.listener.v3.Listener map[test-service:{}] from nodeID "aa24c8dc-da7e-4686-a1ef-bb1f695c367e", version "1"
            logging.go:30: respond type.googleapis.com/envoy.config.route.v3.RouteConfiguration (requested [route-test-service]) version "" with version "1" and resources [route-test-service]
            logging.go:30: open watch 2 for type.googleapis.com/envoy.config.route.v3.RouteConfiguration map[route-test-service:{}] from nodeID "aa24c8dc-da7e-4686-a1ef-bb1f695c367e", version "1"
            logging.go:30: respond type.googleapis.com/envoy.config.cluster.v3.Cluster (requested [cluster-test-service]) version "" with version "1" and resources [cluster-test-service]
            logging.go:30: open watch 3 for type.googleapis.com/envoy.config.cluster.v3.Cluster map[cluster-test-service:{}] from nodeID "aa24c8dc-da7e-4686-a1ef-bb1f695c367e", version "1"
            logging.go:30: respond type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment (requested [endpoints-test-service]) version "" with version "1" and resources [endpoints-test-service]
            logging.go:30: open watch 4 for type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment map[endpoints-test-service:{}] from nodeID "aa24c8dc-da7e-4686-a1ef-bb1f695c367e", version "1"
            ext_proc_ext_test.go:1358: EmptyCall() returned error message "rpc error: code = Internal desc = extproc: failed to send client headers to external processor server: EOF", want it to contain "abrupt stream failure"
        tlogger.go:133: INFO server.go:734 [core] [Server #484] Server created  (t=+11.239764ms)
        tlogger.go:133: INFO server.go:734 [core] [Server #485] Server created  (t=+11.38635ms)
        tlogger.go:133: INFO server.go:734 [core] [Server #486] Server created  (t=+11.541309ms)
        tlogger.go:133: INFO clientconn.go:1823 [core] original dial target is: "xds:///test-service"  (t=+12.277485ms)
        tlogger.go:133: INFO clientconn.go:514 [core] [Channel #487] Channel created for target "xds:///test-service"  (t=+12.302052ms)
        tlogger.go:133: INFO clientconn.go:245 [core] [Channel #487] parsed dial target is: resolver.Target{URL:url.URL{Scheme:"xds", Opaque:"", User:(*url.Userinfo)(nil), Host:"", Path:"/test-service", Fragment:"", RawQuery:"", RawPath:"", RawFragment:"", ForceQuery:false, OmitHost:false}}  (t=+12.32828ms)
        tlogger.go:133: INFO clientconn.go:246 [core] [Channel #487] Channel authority set to "test-service"  (t=+12.342511ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #487] Channel Connectivity change to CONNECTING  (t=+12.370733ms)
        tlogger.go:133: INFO server.go:930 [core] [Server #484 ListenSocket #488] ListenSocket created  (t=+12.433606ms)
        tlogger.go:133: INFO server.go:930 [core] [Server #485 ListenSocket #489] ListenSocket created  (t=+12.481727ms)
        tlogger.go:133: INFO server.go:930 [core] [Server #486 ListenSocket #490] ListenSocket created  (t=+12.523178ms)
        tlogger.go:133: INFO pool.go:309 [xds] xDS node ID: 4e62bae6-c407-42f0-93d3-b7bcd8ff0f3e  (t=+12.709322ms)
        tlogger.go:133: INFO xds_resolver.go:158 [xds] [xds-resolver 0x2480d2b72630] Creating resolver for target: xds:///test-service  (t=+12.737674ms)
        tlogger.go:133: INFO clientconn.go:1823 [core] original dial target is: "passthrough:///127.0.0.1:38359"  (t=+12.807968ms)
        tlogger.go:133: INFO clientconn.go:514 [core] [Channel #491] Channel created for target "passthrough:///127.0.0.1:38359"  (t=+12.831783ms)
        tlogger.go:133: INFO clientconn.go:245 [core] [Channel #491] parsed dial target is: resolver.Target{URL:url.URL{Scheme:"passthrough", Opaque:"", User:(*url.Userinfo)(nil), Host:"", Path:"/127.0.0.1:38359", Fragment:"", RawQuery:"", RawPath:"", RawFragment:"", ForceQuery:false, OmitHost:false}}  (t=+12.85671ms)
        tlogger.go:133: INFO clientconn.go:246 [core] [Channel #491] Channel authority set to "127.0.0.1:38359"  (t=+12.871932ms)
        tlogger.go:133: INFO clientconn.go:419 [core] [Channel #487] Channel exiting idle mode  (t=+12.92456ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #491] Channel Connectivity change to CONNECTING  (t=+12.975956ms)
        tlogger.go:133: INFO balancer_wrapper.go:121 [core] [Channel #491] Channel switches to new LB policy "pick_first"  (t=+13.010086ms)
        tlogger.go:133: INFO clientconn.go:922 [core] [Channel #491 SubChannel #492] Subchannel created  (t=+13.054902ms)
        tlogger.go:133: INFO clientconn.go:419 [core] [Channel #491] Channel exiting idle mode  (t=+13.08078ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #491 SubChannel #492] Subchannel Connectivity change to CONNECTING  (t=+13.130854ms)
        tlogger.go:133: INFO clientconn.go:1470 [core] [Channel #491 SubChannel #492] Subchannel picks a new address "127.0.0.1:38359" to connect  (t=+13.146628ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #491 SubChannel #492] Subchannel Connectivity change to READY  (t=+13.608499ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #491] Channel Connectivity change to READY  (t=+13.643571ms)
        tlogger.go:133: INFO clientconn.go:1823 [core] original dial target is: "127.0.0.1:37483"  (t=+14.927495ms)
        tlogger.go:133: INFO clientconn.go:514 [core] [Channel #495] Channel created for target "127.0.0.1:37483"  (t=+14.961536ms)
        tlogger.go:133: INFO clientconn.go:245 [core] [Channel #495] parsed dial target is: resolver.Target{URL:url.URL{Scheme:"dns", Opaque:"", User:(*url.Userinfo)(nil), Host:"", Path:"/127.0.0.1:37483", Fragment:"", RawQuery:"", RawPath:"", RawFragment:"", ForceQuery:false, OmitHost:false}}  (t=+14.981665ms)
        tlogger.go:133: INFO clientconn.go:246 [core] [Channel #495] Channel authority set to "127.0.0.1:37483"  (t=+14.995626ms)
        tlogger.go:133: INFO balancer_wrapper.go:121 [core] [Channel #487] Channel switches to new LB policy "xds_cluster_manager_experimental"  (t=+15.07319ms)
        tlogger.go:133: INFO clustermanager.go:58 [xds] [xds-cluster-manager-lb 0x2480d2bf8920] Created  (t=+15.098057ms)
        tlogger.go:133: INFO balancergroup.go:274 [xds] [xds-cluster-manager-lb 0x2480d2bf8920] Adding child policy of type "cds_experimental" for child "cluster:cluster-test-service"  (t=+15.121141ms)
        tlogger.go:133: INFO balancergroup.go:102 [xds] [xds-cluster-manager-lb 0x2480d2bf8920] Creating child policy of type "cds_experimental" for child "cluster:cluster-test-service"  (t=+15.135482ms)
        tlogger.go:133: INFO cdsbalancer.go:90 [xds] [cds-lb 0x2480d29501c0] Created  (t=+15.151295ms)
        tlogger.go:133: INFO cdsbalancer.go:167 [xds] [cds-lb 0x2480d29501c0] Received balancer config update: {
              "LoadBalancingConfig": null,
              "cluster": "cluster-test-service",
              "isDynamic": false
            }  (t=+15.171115ms)
        tlogger.go:133: INFO balancer.go:76 [xds] [priority-lb 0x2480d2d6d030] Created  (t=+15.226476ms)
        tlogger.go:133: INFO balancergroup.go:528 [xds] [xds-cluster-manager-lb 0x2480d2bf8920] Balancer state update from child cluster:cluster-test-service, new state: {ConnectivityState:CONNECTING Picker:0x2480d2c8b830}  (t=+15.37145ms)
        tlogger.go:133: INFO balancerstateaggregator.go:147 [xds] [xds-cluster-manager-lb 0x2480d2bf8920] State update from sub-balancer "cluster:cluster-test-service": {ConnectivityState:CONNECTING Picker:0x2480d2c8b830}  (t=+15.394324ms)
        tlogger.go:133: INFO balancergroup.go:274 [xds] [priority-lb 0x2480d2d6d030] Adding child policy of type "outlier_detection_experimental" for child "priority-0-0"  (t=+15.406692ms)
        tlogger.go:133: INFO balancergroup.go:102 [xds] [priority-lb 0x2480d2d6d030] Creating child policy of type "outlier_detection_experimental" for child "priority-0-0"  (t=+15.41882ms)
        tlogger.go:133: INFO balancer.go:94 [xds] [outlier-detection-lb 0x2480d2d16c30] Created  (t=+15.434784ms)
        tlogger.go:133: INFO clusterimpl.go:88 [xds] [xds-cluster-impl-lb 0x2480d2ac6b40] Created  (t=+15.454293ms)
        tlogger.go:133: INFO clusterimpl.go:100 [xds] [xds-cluster-impl-lb 0x2480d2ac6b40] xDS credentials in use: false  (t=+15.468554ms)
        tlogger.go:133: INFO weightedtarget.go:65 [xds] [weighted-target-lb 0x2480d2bf8d80] Created  (t=+15.521321ms)
        tlogger.go:133: INFO balancer.go:88 [xds] [wrrlocality-lb 0x2480d2d11770] Created  (t=+15.537756ms)
        tlogger.go:133: INFO balancergroup.go:274 [xds] [weighted-target-lb 0x2480d2bf8d80] Adding child policy of type "round_robin" for child "{region=\"region-1\", zone=\"zone-1\", sub_zone=\"subzone-1\"}"  (t=+15.570664ms)
        tlogger.go:133: INFO balancergroup.go:102 [xds] [weighted-target-lb 0x2480d2bf8d80] Creating child policy of type "round_robin" for child "{region=\"region-1\", zone=\"zone-1\", sub_zone=\"subzone-1\"}"  (t=+15.587329ms)
        tlogger.go:133: INFO roundrobin.go:56 [roundrobin] [0x2480d2d11aa0] Created  (t=+15.605285ms)
        tlogger.go:133: INFO clientconn.go:922 [core] [Channel #487 SubChannel #496] Subchannel created  (t=+15.642911ms)
        tlogger.go:133: INFO balancergroup.go:528 [xds] [weighted-target-lb 0x2480d2bf8d80] Balancer state update from child {region="region-1", zone="zone-1", sub_zone="subzone-1"}, new state: {ConnectivityState:CONNECTING Picker:0x2480d2906400}  (t=+15.676851ms)
        tlogger.go:133: INFO aggregator.go:253 [xds] [weighted-target-lb 0x2480d2bf8d80] Child pickers with config: map[{region="region-1", zone="zone-1", sub_zone="subzone-1"}:weight:1,picker:0x2480d2906400,state:CONNECTING,stateToAggregate:CONNECTING]  (t=+15.701037ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #487 SubChannel #496] Subchannel Connectivity change to CONNECTING  (t=+15.736679ms)
        tlogger.go:133: INFO clientconn.go:1470 [core] [Channel #487 SubChannel #496] Subchannel picks a new address "localhost:33973" to connect  (t=+15.75105ms)
        tlogger.go:133: INFO balancergroup.go:528 [xds] [priority-lb 0x2480d2d6d030] Balancer state update from child priority-0-0, new state: {ConnectivityState:CONNECTING Picker:0x2480d2e7ec30}  (t=+15.937215ms)
        tlogger.go:133: INFO balancergroup.go:528 [xds] [xds-cluster-manager-lb 0x2480d2bf8920] Balancer state update from child cluster:cluster-test-service, new state: {ConnectivityState:CONNECTING Picker:0x2480d2e7ec30}  (t=+15.952347ms)
        tlogger.go:133: INFO balancerstateaggregator.go:147 [xds] [xds-cluster-manager-lb 0x2480d2bf8920] State update from sub-balancer "cluster:cluster-test-service": {ConnectivityState:CONNECTING Picker:0x2480d2e7ec30}  (t=+15.964215ms)
        tlogger.go:133: INFO balancerstateaggregator.go:191 [xds] [xds-cluster-manager-lb 0x2480d2bf8920] Child pickers: map[cluster:cluster-test-service:picker:0x2480d2e7ec30,state:CONNECTING,stateToAggregate:CONNECTING]  (t=+15.980639ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #495] Channel Connectivity change to CONNECTING  (t=+16.027038ms)
        tlogger.go:133: INFO balancer_wrapper.go:121 [core] [Channel #495] Channel switches to new LB policy "pick_first"  (t=+16.056761ms)
        tlogger.go:133: INFO clientconn.go:922 [core] [Channel #495 SubChannel #497] Subchannel created  (t=+16.084072ms)
        tlogger.go:133: INFO clientconn.go:419 [core] [Channel #495] Channel exiting idle mode  (t=+16.100186ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #495 SubChannel #497] Subchannel Connectivity change to CONNECTING  (t=+16.13053ms)
        tlogger.go:133: INFO clientconn.go:1470 [core] [Channel #495 SubChannel #497] Subchannel picks a new address "127.0.0.1:37483" to connect  (t=+16.143389ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #495 SubChannel #497] Subchannel Connectivity change to READY  (t=+16.636547ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #495] Channel Connectivity change to READY  (t=+16.662275ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #487] Channel Connectivity change to SHUTDOWN  (t=+16.933505ms)
        tlogger.go:133: INFO resolver_wrapper.go:112 [core] [Channel #487] Closing the name resolver  (t=+16.947366ms)
        tlogger.go:133: INFO balancer_wrapper.go:159 [core] [Channel #487] ccBalancerWrapper: closing  (t=+16.987295ms)
        tlogger.go:133: INFO clusterimpl.go:521 [xds] [xds-cluster-impl-lb 0x2480d2ac6b40] Shutdown  (t=+17.049046ms)
        tlogger.go:133: INFO balancergroup.go:351 [xds] [priority-lb 0x2480d2d6d030] Removing child policy for child "priority-0-0"  (t=+17.068124ms)
        tlogger.go:133: INFO cdsbalancer.go:416 [xds] [cds-lb 0x2480d29501c0] Shutdown  (t=+17.080883ms)
        tlogger.go:133: INFO clustermanager.go:192 [xds] [xds-cluster-manager-lb 0x2480d2bf8920] Shutdown  (t=+17.091929ms)
        tlogger.go:129: WARNING ads_stream.go:410 [xds] [xds-client 0x2480d2b5c960] [xds-channel 0x2480d2bf2c30] [ads-stream 0x2480d2b5ca00] Sending ADS request for type "type.googleapis.com/envoy.config.listener.v3.Listener", resources: [], version: "1", nonce: "1" failed: EOF  (t=+17.11276ms)
        tlogger.go:129: WARNING ads_stream.go:452 [xds] [xds-client 0x2480d2b5c960] [xds-channel 0x2480d2bf2c30] [ads-stream 0x2480d2b5ca00] ADS stream closed: rpc error: code = Canceled desc = context canceled  (t=+17.137907ms)
        tlogger.go:133: INFO ads_stream.go:165 [xds] [xds-client 0x2480d2b5c960] [xds-channel 0x2480d2bf2c30] [ads-stream 0x2480d2b5ca00] Shutdown ADS stream  (t=+17.154411ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #491] Channel Connectivity change to SHUTDOWN  (t=+17.170164ms)
        tlogger.go:133: INFO resolver_wrapper.go:112 [core] [Channel #491] Closing the name resolver  (t=+17.184916ms)
        tlogger.go:133: INFO balancer_wrapper.go:159 [core] [Channel #491] ccBalancerWrapper: closing  (t=+17.198847ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #491 SubChannel #492] Subchannel Connectivity change to SHUTDOWN  (t=+17.231915ms)
        tlogger.go:133: INFO clientconn.go:1696 [core] [Channel #491 SubChannel #492] Subchannel deleted  (t=+17.245035ms)
        tlogger.go:133: INFO clientconn.go:1231 [core] [Channel #491] Channel deleted  (t=+17.30935ms)
        tlogger.go:133: INFO channel.go:143 [xds] [xds-client 0x2480d2b5c960] [xds-channel 0x2480d2bf2c30] Shutdown  (t=+17.324692ms)
        tlogger.go:133: INFO xdsclient.go:210 [xds] [xds-client 0x2480d2b5c960] Shutdown  (t=+17.347155ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #495] Channel Connectivity change to SHUTDOWN  (t=+17.361897ms)
        tlogger.go:133: INFO resolver_wrapper.go:112 [core] [Channel #495] Closing the name resolver  (t=+17.373685ms)
        tlogger.go:133: INFO balancer_wrapper.go:159 [core] [Channel #495] ccBalancerWrapper: closing  (t=+17.387295ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #495 SubChannel #497] Subchannel Connectivity change to SHUTDOWN  (t=+17.418741ms)
        tlogger.go:133: INFO clientconn.go:1696 [core] [Channel #495 SubChannel #497] Subchannel deleted  (t=+17.431119ms)
        tlogger.go:133: INFO clientconn.go:1231 [core] [Channel #495] Channel deleted  (t=+17.47907ms)
        tlogger.go:133: INFO xds_resolver.go:296 [xds] [xds-resolver 0x2480d2b72630] Shutdown  (t=+17.49253ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #487 SubChannel #496] Subchannel Connectivity change to SHUTDOWN  (t=+17.514552ms)
        tlogger.go:133: INFO clientconn.go:1696 [core] [Channel #487 SubChannel #496] Subchannel deleted  (t=+17.52677ms)
        tlogger.go:133: INFO clientconn.go:1231 [core] [Channel #487] Channel deleted  (t=+17.538037ms)
        tlogger.go:133: INFO server.go:866 [core] [Server #485 ListenSocket #489] ListenSocket deleted  (t=+17.570936ms)
        tlogger.go:129: WARNING clientconn.go:1533 [core] [Channel #487 SubChannel #496] grpc: addrConn.createTransport failed to connect to {Addr: "localhost:33973", ServerName: "test-service", Attributes: {"<%!p(xds.hostnameKey={})>": "" , "<%!p(xds.handshakeAttrKey={})>": "<0x2480d2ac6bb0>" , "<%!p(xds.handshakeClusterNameKey={})>": "cluster-test-service" }, BalancerAttributes: {"<%!p(weight.attributeKey={})>": "<%!p(weight.EndpointInfo={2147483648})>" , "<%!p(wrrlocality.attributeKey={})>": "Locality Weight: 1" , "<%!p(xds.localityKeyType=grpc.xds.internal.address.locality)>": "<%!p(clients.Locality={region-1 zone-1 subzone-1})>" , "<%!p(hierarchy.pathKeyType=grpc.internal.address.hierarchical_path)>": "<0x2480d2bf8ac0>" }}. Err: connection error: desc = "transport: Error while dialing: dial tcp [::1]:33973: connect: connection refused"  (t=+17.846652ms)
        tlogger.go:133: INFO server.go:1043 [core] [Server #485] grpc: Server.Serve failed to create ServerTransport: connection error: desc = "transport: http2Server.HandleStreams failed to receive the preface from client: read tcp 127.0.0.1:33973->127.0.0.1:56986: read: connection reset by peer"  (t=+17.939539ms)
        tlogger.go:133: INFO server.go:866 [core] [Server #486 ListenSocket #490] ListenSocket deleted  (t=+17.980139ms)
        tlogger.go:133: INFO server.go:866 [core] [Server #484 ListenSocket #488] ListenSocket deleted  (t=+18.009252ms)
        tlogger.go:133: INFO server.go:734 [core] [Server #501] Server created  (t=+18.161747ms)
        tlogger.go:133: INFO server.go:734 [core] [Server #502] Server created  (t=+18.267473ms)
        tlogger.go:133: INFO server.go:734 [core] [Server #503] Server created  (t=+18.384075ms)
        tlogger.go:133: INFO clientconn.go:1823 [core] original dial target is: "xds:///test-service"  (t=+19.012783ms)
        tlogger.go:133: INFO clientconn.go:514 [core] [Channel #504] Channel created for target "xds:///test-service"  (t=+19.029027ms)
        tlogger.go:133: INFO clientconn.go:245 [core] [Channel #504] parsed dial target is: resolver.Target{URL:url.URL{Scheme:"xds", Opaque:"", User:(*url.Userinfo)(nil), Host:"", Path:"/test-service", Fragment:"", RawQuery:"", RawPath:"", RawFragment:"", ForceQuery:false, OmitHost:false}}  (t=+19.047284ms)
        tlogger.go:133: INFO clientconn.go:246 [core] [Channel #504] Channel authority set to "test-service"  (t=+19.058851ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #504] Channel Connectivity change to CONNECTING  (t=+19.080783ms)
        tlogger.go:133: INFO server.go:930 [core] [Server #501 ListenSocket #505] ListenSocket created  (t=+19.124928ms)
        tlogger.go:133: INFO server.go:930 [core] [Server #502 ListenSocket #506] ListenSocket created  (t=+19.163445ms)
        tlogger.go:133: INFO server.go:930 [core] [Server #503 ListenSocket #507] ListenSocket created  (t=+19.196214ms)
        tlogger.go:133: INFO pool.go:309 [xds] xDS node ID: 40577923-453a-49b4-ae1e-45d637f670d8  (t=+19.333717ms)
        tlogger.go:133: INFO xds_resolver.go:158 [xds] [xds-resolver 0x2480d2cdb8c0] Creating resolver for target: xds:///test-service  (t=+19.351383ms)
        tlogger.go:133: INFO clientconn.go:1823 [core] original dial target is: "passthrough:///127.0.0.1:34441"  (t=+19.385543ms)
        tlogger.go:133: INFO clientconn.go:514 [core] [Channel #508] Channel created for target "passthrough:///127.0.0.1:34441"  (t=+19.402198ms)
        tlogger.go:133: INFO clientconn.go:245 [core] [Channel #508] parsed dial target is: resolver.Target{URL:url.URL{Scheme:"passthrough", Opaque:"", User:(*url.Userinfo)(nil), Host:"", Path:"/127.0.0.1:34441", Fragment:"", RawQuery:"", RawPath:"", RawFragment:"", ForceQuery:false, OmitHost:false}}  (t=+19.418792ms)
        tlogger.go:133: INFO clientconn.go:246 [core] [Channel #508] Channel authority set to "127.0.0.1:34441"  (t=+19.432012ms)
        tlogger.go:133: INFO clientconn.go:419 [core] [Channel #504] Channel exiting idle mode  (t=+19.470368ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #508] Channel Connectivity change to CONNECTING  (t=+19.506201ms)
        tlogger.go:133: INFO balancer_wrapper.go:121 [core] [Channel #508] Channel switches to new LB policy "pick_first"  (t=+19.529095ms)
        tlogger.go:133: INFO clientconn.go:922 [core] [Channel #508 SubChannel #509] Subchannel created  (t=+19.558719ms)
        tlogger.go:133: INFO clientconn.go:419 [core] [Channel #508] Channel exiting idle mode  (t=+19.574682ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #508 SubChannel #509] Subchannel Connectivity change to CONNECTING  (t=+19.615032ms)
        tlogger.go:133: INFO clientconn.go:1470 [core] [Channel #508 SubChannel #509] Subchannel picks a new address "127.0.0.1:34441" to connect  (t=+19.628341ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #508 SubChannel #509] Subchannel Connectivity change to READY  (t=+19.901905ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #508] Channel Connectivity change to READY  (t=+19.936836ms)
        tlogger.go:133: INFO clientconn.go:1823 [core] original dial target is: "127.0.0.1:44401"  (t=+20.852507ms)
        tlogger.go:133: INFO clientconn.go:514 [core] [Channel #512] Channel created for target "127.0.0.1:44401"  (t=+20.884174ms)
        tlogger.go:133: INFO clientconn.go:245 [core] [Channel #512] parsed dial target is: resolver.Target{URL:url.URL{Scheme:"dns", Opaque:"", User:(*url.Userinfo)(nil), Host:"", Path:"/127.0.0.1:44401", Fragment:"", RawQuery:"", RawPath:"", RawFragment:"", ForceQuery:false, OmitHost:false}}  (t=+20.902661ms)
        tlogger.go:133: INFO clientconn.go:246 [core] [Channel #512] Channel authority set to "127.0.0.1:44401"  (t=+20.916862ms)
        tlogger.go:133: INFO balancer_wrapper.go:121 [core] [Channel #504] Channel switches to new LB policy "xds_cluster_manager_experimental"  (t=+20.989119ms)
        tlogger.go:133: INFO clustermanager.go:58 [xds] [xds-cluster-manager-lb 0x2480d27e7840] Created  (t=+21.012514ms)
        tlogger.go:133: INFO balancergroup.go:274 [xds] [xds-cluster-manager-lb 0x2480d27e7840] Adding child policy of type "cds_experimental" for child "cluster:cluster-test-service"  (t=+21.027195ms)
        tlogger.go:133: INFO balancergroup.go:102 [xds] [xds-cluster-manager-lb 0x2480d27e7840] Creating child policy of type "cds_experimental" for child "cluster:cluster-test-service"  (t=+21.039383ms)
        tlogger.go:133: INFO cdsbalancer.go:90 [xds] [cds-lb 0x2480d2c0a700] Created  (t=+21.055527ms)
        tlogger.go:133: INFO cdsbalancer.go:167 [xds] [cds-lb 0x2480d2c0a700] Received balancer config update: {
              "LoadBalancingConfig": null,
              "cluster": "cluster-test-service",
              "isDynamic": false
            }  (t=+21.074034ms)
        tlogger.go:133: INFO balancer.go:76 [xds] [priority-lb 0x2480d2ac1d50] Created  (t=+21.125971ms)
        tlogger.go:133: INFO balancergroup.go:528 [xds] [xds-cluster-manager-lb 0x2480d27e7840] Balancer state update from child cluster:cluster-test-service, new state: {ConnectivityState:CONNECTING Picker:0x2480d266f540}  (t=+21.264075ms)
        tlogger.go:133: INFO balancerstateaggregator.go:147 [xds] [xds-cluster-manager-lb 0x2480d27e7840] State update from sub-balancer "cluster:cluster-test-service": {ConnectivityState:CONNECTING Picker:0x2480d266f540}  (t=+21.285336ms)
        tlogger.go:133: INFO balancergroup.go:274 [xds] [priority-lb 0x2480d2ac1d50] Adding child policy of type "outlier_detection_experimental" for child "priority-0-0"  (t=+21.295641ms)
        tlogger.go:133: INFO balancergroup.go:102 [xds] [priority-lb 0x2480d2ac1d50] Creating child policy of type "outlier_detection_experimental" for child "priority-0-0"  (t=+21.306307ms)
        tlogger.go:133: INFO balancer.go:94 [xds] [outlier-detection-lb 0x2480d2abe780] Created  (t=+21.321029ms)
        tlogger.go:133: INFO clusterimpl.go:88 [xds] [xds-cluster-impl-lb 0x2480d2ba3180] Created  (t=+21.341549ms)
        tlogger.go:133: INFO clusterimpl.go:100 [xds] [xds-cluster-impl-lb 0x2480d2ba3180] xDS credentials in use: false  (t=+21.354548ms)
        tlogger.go:133: INFO weightedtarget.go:65 [xds] [weighted-target-lb 0x2480d27e7ca0] Created  (t=+21.40288ms)
        tlogger.go:133: INFO balancer.go:88 [xds] [wrrlocality-lb 0x2480d2c44c90] Created  (t=+21.418432ms)
        tlogger.go:133: INFO balancergroup.go:274 [xds] [weighted-target-lb 0x2480d27e7ca0] Adding child policy of type "round_robin" for child "{region=\"region-1\", zone=\"zone-1\", sub_zone=\"subzone-1\"}"  (t=+21.450279ms)
        tlogger.go:133: INFO balancergroup.go:102 [xds] [weighted-target-lb 0x2480d27e7ca0] Creating child policy of type "round_robin" for child "{region=\"region-1\", zone=\"zone-1\", sub_zone=\"subzone-1\"}"  (t=+21.466243ms)
        tlogger.go:133: INFO roundrobin.go:56 [roundrobin] [0x2480d2c44fc0] Created  (t=+21.484199ms)
        tlogger.go:133: INFO clientconn.go:922 [core] [Channel #504 SubChannel #513] Subchannel created  (t=+21.517849ms)
        tlogger.go:133: INFO balancergroup.go:528 [xds] [weighted-target-lb 0x2480d27e7ca0] Balancer state update from child {region="region-1", zone="zone-1", sub_zone="subzone-1"}, new state: {ConnectivityState:CONNECTING Picker:0x2480d2956740}  (t=+21.548865ms)
        tlogger.go:133: INFO aggregator.go:253 [xds] [weighted-target-lb 0x2480d27e7ca0] Child pickers with config: map[{region="region-1", zone="zone-1", sub_zone="subzone-1"}:weight:1,picker:0x2480d2956740,state:CONNECTING,stateToAggregate:CONNECTING]  (t=+21.570607ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #504 SubChannel #513] Subchannel Connectivity change to CONNECTING  (t=+21.60006ms)
        tlogger.go:133: INFO clientconn.go:1470 [core] [Channel #504 SubChannel #513] Subchannel picks a new address "localhost:46535" to connect  (t=+21.6135ms)
        tlogger.go:133: INFO balancergroup.go:528 [xds] [priority-lb 0x2480d2ac1d50] Balancer state update from child priority-0-0, new state: {ConnectivityState:CONNECTING Picker:0x2480d2b8d140}  (t=+21.824181ms)
        tlogger.go:133: INFO balancergroup.go:528 [xds] [xds-cluster-manager-lb 0x2480d27e7840] Balancer state update from child cluster:cluster-test-service, new state: {ConnectivityState:CONNECTING Picker:0x2480d2b8d140}  (t=+21.840645ms)
        tlogger.go:133: INFO balancerstateaggregator.go:147 [xds] [xds-cluster-manager-lb 0x2480d27e7840] State update from sub-balancer "cluster:cluster-test-service": {ConnectivityState:CONNECTING Picker:0x2480d2b8d140}  (t=+21.853194ms)
        tlogger.go:133: INFO balancerstateaggregator.go:191 [xds] [xds-cluster-manager-lb 0x2480d27e7840] Child pickers: map[cluster:cluster-test-service:picker:0x2480d2b8d140,state:CONNECTING,stateToAggregate:CONNECTING]  (t=+21.869137ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #512] Channel Connectivity change to CONNECTING  (t=+21.913293ms)
        tlogger.go:133: INFO balancer_wrapper.go:121 [core] [Channel #512] Channel switches to new LB policy "pick_first"  (t=+21.941394ms)
        tlogger.go:133: INFO clientconn.go:922 [core] [Channel #512 SubChannel #514] Subchannel created  (t=+21.970327ms)
        tlogger.go:133: INFO clientconn.go:419 [core] [Channel #512] Channel exiting idle mode  (t=+21.988243ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #512 SubChannel #514] Subchannel Connectivity change to CONNECTING  (t=+22.020481ms)
        tlogger.go:133: INFO clientconn.go:1470 [core] [Channel #512 SubChannel #514] Subchannel picks a new address "127.0.0.1:44401" to connect  (t=+22.032549ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #512 SubChannel #514] Subchannel Connectivity change to READY  (t=+22.436415ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #512] Channel Connectivity change to READY  (t=+22.458307ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #504] Channel Connectivity change to SHUTDOWN  (t=+22.86039ms)
        tlogger.go:133: INFO resolver_wrapper.go:112 [core] [Channel #504] Closing the name resolver  (t=+22.874761ms)
        tlogger.go:133: INFO balancer_wrapper.go:159 [core] [Channel #504] ccBalancerWrapper: closing  (t=+22.887931ms)
        tlogger.go:133: INFO clusterimpl.go:521 [xds] [xds-cluster-impl-lb 0x2480d2ba3180] Shutdown  (t=+22.943423ms)
        tlogger.go:133: INFO balancergroup.go:351 [xds] [priority-lb 0x2480d2ac1d50] Removing child policy for child "priority-0-0"  (t=+22.96167ms)
        tlogger.go:133: INFO cdsbalancer.go:416 [xds] [cds-lb 0x2480d2c0a700] Shutdown  (t=+22.973036ms)
        tlogger.go:133: INFO clustermanager.go:192 [xds] [xds-cluster-manager-lb 0x2480d27e7840] Shutdown  (t=+22.982911ms)
        tlogger.go:129: WARNING ads_stream.go:410 [xds] [xds-client 0x2480d29f7220] [xds-channel 0x2480d2d14e60] [ads-stream 0x2480d29f72c0] Sending ADS request for type "type.googleapis.com/envoy.config.listener.v3.Listener", resources: [], version: "1", nonce: "1" failed: EOF  (t=+23.002259ms)
        tlogger.go:129: WARNING ads_stream.go:452 [xds] [xds-client 0x2480d29f7220] [xds-channel 0x2480d2d14e60] [ads-stream 0x2480d29f72c0] ADS stream closed: rpc error: code = Canceled desc = context canceled  (t=+23.021428ms)
        tlogger.go:133: INFO ads_stream.go:165 [xds] [xds-client 0x2480d29f7220] [xds-channel 0x2480d2d14e60] [ads-stream 0x2480d29f72c0] Shutdown ADS stream  (t=+23.034837ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #508] Channel Connectivity change to SHUTDOWN  (t=+23.049189ms)
        tlogger.go:133: INFO resolver_wrapper.go:112 [core] [Channel #508] Closing the name resolver  (t=+23.061457ms)
        tlogger.go:133: INFO balancer_wrapper.go:159 [core] [Channel #508] ccBalancerWrapper: closing  (t=+23.075357ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #508 SubChannel #509] Subchannel Connectivity change to SHUTDOWN  (t=+23.105171ms)
        tlogger.go:133: INFO clientconn.go:1696 [core] [Channel #508 SubChannel #509] Subchannel deleted  (t=+23.116127ms)
        tlogger.go:133: INFO clientconn.go:1231 [core] [Channel #508] Channel deleted  (t=+23.186481ms)
        tlogger.go:133: INFO channel.go:143 [xds] [xds-client 0x2480d29f7220] [xds-channel 0x2480d2d14e60] Shutdown  (t=+23.201323ms)
        tlogger.go:133: INFO xdsclient.go:210 [xds] [xds-client 0x2480d29f7220] Shutdown  (t=+23.221303ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #512] Channel Connectivity change to SHUTDOWN  (t=+23.235183ms)
        tlogger.go:133: INFO resolver_wrapper.go:112 [core] [Channel #512] Closing the name resolver  (t=+23.245729ms)
        tlogger.go:133: INFO balancer_wrapper.go:159 [core] [Channel #512] ccBalancerWrapper: closing  (t=+23.257316ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #512 SubChannel #514] Subchannel Connectivity change to SHUTDOWN  (t=+23.284696ms)
        tlogger.go:133: INFO clientconn.go:1696 [core] [Channel #512 SubChannel #514] Subchannel deleted  (t=+23.294861ms)
        tlogger.go:133: INFO clientconn.go:1231 [core] [Channel #512] Channel deleted  (t=+23.337564ms)
        tlogger.go:133: INFO xds_resolver.go:296 [xds] [xds-resolver 0x2480d2cdb8c0] Shutdown  (t=+23.351074ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #504 SubChannel #513] Subchannel Connectivity change to SHUTDOWN  (t=+23.372235ms)
        tlogger.go:133: INFO clientconn.go:1696 [core] [Channel #504 SubChannel #513] Subchannel deleted  (t=+23.384704ms)
        tlogger.go:133: INFO clientconn.go:1231 [core] [Channel #504] Channel deleted  (t=+23.394789ms)
        tlogger.go:133: INFO server.go:866 [core] [Server #502 ListenSocket #506] ListenSocket deleted  (t=+23.448448ms)
        tlogger.go:133: INFO server.go:866 [core] [Server #503 ListenSocket #507] ListenSocket deleted  (t=+23.502227ms)
        tlogger.go:133: INFO server.go:866 [core] [Server #501 ListenSocket #505] ListenSocket deleted  (t=+23.562256ms)
FAIL
```

--- a/internal/xds/httpfilter/extproc/ext_proc.go
+++ b/internal/xds/httpfilter/extproc/ext_proc.go
@@ -313,6 +313,9 @@ func (i *clientInterceptor) NewStream(ctx context.Context, ri resolver.RPCInfo, 
 			},
 		}
 		if err = cs.procStream.Send(headerReq); err != nil {
+			if err == io.EOF {
+				_, err = cs.procStream.Recv()
+			}
 			return cs.handleProcStreamInitError(fmt.Errorf("failed to send client headers to external processor server: %v", err), newStream, opts)
 		}
 	} else {

